### ISSUE TITLE
Unexpected behavior with adding TXT records to DynuDNS 

### ISSUE THREAD

**Issue by NickMagpie** [acmesh-official/acme.sh#2817]
I have a Linux system running Debian 9.12 (Stretch)  with Apache/2.4.25 (Debian) that is used as proxy server for several applications. I also have domain name and additional sub-domains in .org zone registered through DynuDNS for which I want to install SSL certificates provided by Let's Encrypt.

Let's Encrypt documentation points to **_acme.sh_** as ACME client for DynuDNS, therefore I have installed latest version from GitHub using advanced method. Installation has been completed but existing installation of Apache HTTPD server was not detected.

After acme.sh installation I've executed it to issue certificates for domain and 2 sub-domains in test mode but I've discovered that at some point of execution script had switched from DynuDNS to Cloudflare to validate one of sub-domains. Because Cloudflare has no information about sub-domain script enters loop and I've cancelled script execution. Execution log with obfuscated data is attached.

I've checked content of DNS records kept on the DynuDNS database and have discovered the following (that seems to me suspicious):
1. 'A' records for domain with two 'TXT' records with acme challenge information:
- 'A' record for **my-domain.org**
- 'A' record for ***.my-domain.org**
- 'TXT' record with key _'_acme-challenge.my-domain.org'_
- 'TXT' record with key _'_acme-challenge.subb.my-domain.org'_

2. 'A' record for sub-domain with one 'TXT' records with acme challenge information:
- 'A' record for ***.suba.my-domain.org**
- 'TXT' record with key _'_acme-challenge.sub.my-domain.org'_

3. 'A' record for sub-domain with **no** 'TXT' records 
- 'A' record for **subb.my-domain.org**
- 'A' record for ***.subb.my-domain.org**

'TXT' records mentioned by **_acme.sh_** documentation were created and inserted into DynuDNS database.
In relation to described above I have following questions:
1. What caused script **_acme.sh_** to switch to cloudflare-dns.com for checks? Is it an intended action or caused by a bug?
2. What changes need to be applied to DNS database to repair the content before script re-run?
3. How acme.sh check the presence of the Apache HTTPD server? What is the version of the Apache HTTPD server supported by acme.sh?
4. Is socat required for **_acme.sh_** operation/execution?

Regards,
Nick Sorokin
[acme.issue.dynudns.mydomain.log](https://github.com/acmesh-official/acme.sh/files/4403771/acme.issue.dynudns.mydomain.log)

**Comment by Neilpang** [comment]
>> What caused script acme.sh to switch to cloudflare-dns.com for checks? Is it an intended action or caused by a bug?

No,  it just uses cloudflare to check whether the txt records have propagated.

>> What changes need to be applied to DNS database to repair the content before script re-run?

if there is a bug, please provide log with `--debug 2`


>> How acme.sh check the presence of the Apache HTTPD server?

it uses `apachectl` or `apache2ctl`

>> Is socat required for acme.sh operation/execution?

No, it's only used for standalone mode.

**Comment by NickMagpie** [comment]
Hi Neil,

Thank you for your time spent looking into my problem and explanation of acme.sh logic. I definitely will re-run script with `--debug 2` option but before that I would like to clear the mess the DynuDNS database and also check the Let's Encrypt relevant data. My question - is there a way to check content of Let's Encrypt database with Let's Encrypt UI or acme.sh commands?

Regards,
Nick

**Comment by Neilpang** [comment]
>> is there a way to check content of Let's Encrypt database with Let's Encrypt UI or acme.sh commands?

No.

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.
